@astro/vercel caused build failed when I upgrade to `8.2.1`

### Astro Info

```block
> astro info

[dotenv@17.1.0] injecting env (23) from .env (tip: ⚙️  suppress all logs with { quiet: true })
[dotenv@17.1.0] injecting env (0) from .env (tip: ⚙️  enable debug logging with { debug: true })
[dotenv@17.1.0] injecting env (0) from .env (tip: 🔐 prevent committing .env to code: https://dotenvx.com/precommit)
[dotenv@17.1.0] injecting env (0) from .env (tip: ⚙️  override existing env vars with { override: true })
Astro                    v5.11.0
Node                     v22.17.0
System                   Linux (x64)
Package Manager          pnpm
Output                   server
Adapter                  @astrojs/vercel
Integrations             @astrojs/mdx
                         @astrojs/react
                         product-indexing
                         blog-indexing
                         resource-indexing
                         file-uploading
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug


`astro build` works with the old version `8.1.3`, but when I upgrade `@astro/verce` to `8.2.1`, the build is failed.

Here's the build output:

```
19:20:30 [WARN] [vite] [plugin vite:resolve] Module "node:fs/promises" has been externalized for browser compatibility, imported by "/home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/astro@5.11.0_@types+node@24.0.10_jiti@2.4.2_lightningcss@1.30.1_rollup@4.44.2_tsx@4.20.3_typescript@5.8.3_yaml@2.8.0/node_modules/astro/dist/assets/utils/node/emitAsset.js". See https://vite.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
19:20:30 [WARN] [vite] [plugin vite:resolve] Module "node:path" has been externalized for browser compatibility, imported by "/home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/astro@5.11.0_@types+node@24.0.10_jiti@2.4.2_lightningcss@1.30.1_rollup@4.44.2_tsx@4.20.3_typescript@5.8.3_yaml@2.8.0/node_modules/astro/dist/assets/utils/node/emitAsset.js". See https://vite.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
19:20:30 [WARN] [vite] [plugin vite:resolve] Module "node:url" has been externalized for browser compatibility, imported by "/home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/astro@5.11.0_@types+node@24.0.10_jiti@2.4.2_lightningcss@1.30.1_rollup@4.44.2_tsx@4.20.3_typescript@5.8.3_yaml@2.8.0/node_modules/astro/dist/assets/utils/node/emitAsset.js". See https://vite.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
19:20:30 [WARN] [vite] [plugin vite:resolve] Module "node:path" has been externalized for browser compatibility, imported by "/home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/astro@5.11.0_@types+node@24.0.10_jiti@2.4.2_lightningcss@1.30.1_rollup@4.44.2_tsx@4.20.3_typescript@5.8.3_yaml@2.8.0/node_modules/astro/dist/assets/utils/transformToPath.js". See https://vite.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
19:20:30 [WARN] [vite] [plugin vite:resolve] Module "node:crypto" has been externalized for browser compatibility, imported by "/home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/deterministic-object-hash@2.0.2/node_modules/deterministic-object-hash/dist/index.js". See https://vite.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
19:20:31 [WARN] [vite] [plugin vite:resolve] Module "stream" has been externalized for browser compatibility, imported by "/home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/node-fetch@2.7.0/node_modules/node-fetch/lib/index.mjs". See https://vite.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
19:20:31 [WARN] [vite] [plugin vite:resolve] Module "http" has been externalized for browser compatibility, imported by "/home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/node-fetch@2.7.0/node_modules/node-fetch/lib/index.mjs". See https://vite.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
19:20:31 [WARN] [vite] [plugin vite:resolve] Module "url" has been externalized for browser compatibility, imported by "/home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/node-fetch@2.7.0/node_modules/node-fetch/lib/index.mjs". See https://vite.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
19:20:31 [WARN] [vite] [plugin vite:resolve] Module "https" has been externalized for browser compatibility, imported by "/home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/node-fetch@2.7.0/node_modules/node-fetch/lib/index.mjs". See https://vite.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
19:20:31 [WARN] [vite] [plugin vite:resolve] Module "zlib" has been externalized for browser compatibility, imported by "/home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/node-fetch@2.7.0/node_modules/node-fetch/lib/index.mjs". See https://vite.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details.
19:20:31 [vite] ✓ 5054 modules transformed.
19:20:31 [ERROR] [vite] ✗ Build failed in 5.21s
node_modules/.pnpm/astro@5.11.0_@types+node@24.0.10_jiti@2.4.2_lightningcss@1.30.1_rollup@4.44.2_tsx@4.20.3_typescript@5.8.3_yaml@2.8.0/node_modules/astro/dist/assets/utils/node/emitAsset.js (3:24): "pathToFileURL" is not exported by "__vite-browser-external", imported by "node_modules/.pnpm/astro@5.11.0_@types+node@24.0.10_jiti@2.4.2_lightningcss@1.30.1_rollup@4.44.2_tsx@4.20.3_typescript@5.8.3_yaml@2.8.0/node_modules/astro/dist/assets/utils/node/emitAsset.js".
file: /home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/astro@5.11.0_@types+node@24.0.10_jiti@2.4.2_lightningcss@1.30.1_rollup@4.44.2_tsx@4.20.3_typescript@5.8.3_yaml@2.8.0/node_modules/astro/dist/assets/utils/node/emitAsset.js:3:24

1: import fs from "node:fs/promises";
2: import path from "node:path";
3: import { fileURLToPath, pathToFileURL } from "node:url";
                           ^
4: import { prependForwardSlash, slash } from "../../../core/path.js";
5: import { imageMetadata } from "../metadata.js";

  Location:
    /home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/astro@5.11.0_@types+node@24.0.10_jiti@2.4.2_lightningcss@1.30.1_rollup@4.44.2_tsx@4.20.3_typescript@5.8.3_yaml@2.8.0/node_modules/astro/dist/assets/utils/node/emitAsset.js:3:24
  Stack trace:
    at getRollupError (file:///home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/rollup@4.44.2/node_modules/rollup/dist/es/shared/parseAst.js:401:41)
    at Module.error (file:///home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/rollup@4.44.2/node_modules/rollup/dist/es/shared/node-entry.js:16803:16)
    at ModuleScope.findVariable (file:///home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/rollup@4.44.2/node_modules/rollup/dist/es/shared/node-entry.js:14901:39)
    at FunctionBodyScope.findVariable (file:///home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/rollup@4.44.2/node_modules/rollup/dist/es/shared/node-entry.js:5620:38)
    at CallExpression.bind (file:///home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/rollup@4.44.2/node_modules/rollup/dist/es/shared/node-entry.js:2785:23)
    at VariableDeclarator.bind (file:///home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/rollup@4.44.2/node_modules/rollup/dist/es/shared/node-entry.js:2785:23)
    at BlockStatement.bind (file:///home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/rollup@4.44.2/node_modules/rollup/dist/es/shared/node-entry.js:2781:28)
    at Program.bind (file:///home/fengyu/projects/DTeam-Top/blog/node_modules/.pnpm/rollup@4.44.2/node_modules/rollup/dist/es/shared/node-entry.js:2781:28)
 ELIFECYCLE  Command failed with exit code 1.
```

Here's my `astro.config.mjs`:

```ts
import mdx from "@astrojs/mdx";
import react from "@astrojs/react";
import vercel from "@astrojs/vercel";
import tailwindcss from "@tailwindcss/vite";
import { defineConfig } from "astro/config";
import rehypeSanitize, { defaultSchema } from "rehype-sanitize";
import { env } from "./src/env";
import blogIndexing from "./src/integrations/blogIndexing";
import fileUploading from "./src/integrations/fileUploading";
import productIndexing from "./src/integrations/productIndexing";
import resourceIndexing from "./src/integrations/resourceIndexing";

export default defineConfig({
  server: {
    // This will allow all hosts to be used in development. Not only localhost.
    allowedHosts: true,
  },
  output: "server",
  adapter: vercel({
    imageService: true,
  }),
  site: env().SITE_URL,
  markdown: {
    rehypePlugins: [rehypeSanitize(defaultSchema)],
  },
  integrations: [
    mdx({
      rehypePlugins: [rehypeSanitize(defaultSchema)],
    }),
    react({
      include: ["**/react/*", "**/components/*"],
    }),
    productIndexing(),
    blogIndexing(),
    resourceIndexing(),
    fileUploading(),
  ],
  vite: {
    plugins: [tailwindcss()],
  },
  security: {
    checkOrigin: false,
  },
});
```

But very strange when I just comment out `imageService: true,`, like:

```ts
  adapter: vercel({
    // imageService: true,
  }),
```

The build is success.

I don't know how to reproduce this issue, but I try to create a new astro by `pnpm create astro@latest`, no this issue.

This is what I observed:

- `@astro/vercel@8.1.3`: everything works very well
- `@astro/vercel@8.2.1`: build failed with `"pathToFileURL" is not exported by "__vite-browser-external"`
- `@astro/vercel@8.2.1` comment out `imageService: true`, everything works very well

And here's astro version:

```sh
> astro --version


   astro  v5.11.0
```

Please help.

Thanks.

### What's the expected result?

`astro build` should work

### Link to Minimal Reproducible Example

TODO

### Participation

- [ ] I am willing to submit a pull request for this issue.

--- a/.changeset/stale-knives-rhyme.md
+++ b/.changeset/stale-knives-rhyme.md
@@ -1,0 +1,5 @@
+---
+"astro": patch
+---
+
+Fixes a bug that caused some image service builds to fail

--- a/packages/astro/src/assets/vite-plugin-assets.ts
+++ b/packages/astro/src/assets/vite-plugin-assets.ts
@@ -117,9 +117,12 @@ export default function assets({ fs, settings, sync, logger }: Options): vite.Pl
 			config(_, env) {
 				isBuild = env.command === 'build';
 			},
-			async resolveId(id) {
+			async resolveId(id, _importer, options) {
 				if (id === VIRTUAL_SERVICE_ID) {
-					return await this.resolve(settings.config.image.service.entrypoint);
+					if (options?.ssr) {
+						return await this.resolve(settings.config.image.service.entrypoint);
+					}
+					return await this.resolve('astro/assets/services/noop');
 				}
 				if (id === VIRTUAL_MODULE_ID) {
 					return resolvedVirtualModuleId;

